docs/README-develop-node-utils-APIs.md

```diff
@@ -3,36 +3,81 @@
  * [Step 1. YAML Definitions: feature bash-shell](#yaml)
  * [Step 2. Create the node_utils API: feature bash-shell](#api)
  * [Step 3. Create the Minitest: feature bash-shell](#minitest)
  * [Step 4. rubocop / lint: feature bash-shell](#lint)
+ * [Step 5. Build and Install the gem](#gem)
 * [Advanced Example: router eigrp](#complex)
  * [Step 1. YAML Definitions: router eigrp](#comp_yaml)
  * [Step 2. Create the node_utils API: router eigrp](#comp_api)
  * [Step 3. Create the Minitest: router eigrp](#comp_minitest)
  * [Step 4. rubocop / lint: router eigrp](#comp_lint)
+ * [Step 5. Build and Install the gem](#comp_gem)
 
 ## <a name="overview">Overview</a>
 
-This document is a HowTo guide for writing new cisco node_utils APIs. The node_utils APIs act as an interface between the NX-OS CLI and an agent's resource/provider. If written properly the new API will work as a common framework for multiple providers (Puppet, Chef, etc).
+This document is a HowTo guide for writing new cisco_node_utils APIs. The APIs act as an interface between the NX-OS CLI and an agent's resource/provider. If written properly the new API will work as a common framework for multiple providers (Puppet, Chef, etc).
 
-There are multiple components involved when creating new resources. This document focuses on the cisco node_utils API, command reference YAML files, and minitests.
+There are multiple components involved when creating new resources. This document focuses on the cisco_node_utils API, command reference YAML files, and minitests.
 
 ![1](agent_files.png)
 
-## <a name="clone">Start here: Clone the Repo</a>
+## <a name="prerequisites">Before You Begin</a>
 
-First install the code base. Clone the cisco_node_utils repo into a workspace:
+This development guide uses tools that are packaged as gems that need to be installed on your server.
 
 ```bash
-git clone https://github.com/cisco/cisco-network-node-utils.git
+gem install cisco_nxapi
+gem install rubocop
+gem install minitest --version 4.3.2
+```
+
+**NOTE:** If you are working from a server where you don't have admin/root privilages, use the following commands to install the gems and then update your `PATH` to include `~/.gem/ruby/x.x.x/bin`
+
+```bash
+gem install --user-install cisco_nxapi
+gem install --user-install rubocop
+gem install --user-install minitest --version 4.3.2
+```
+
+## <a name="clone">Start here: Fork and Clone the Repo</a>
+
+First [fork](https://help.github.com/articles/fork-a-repo) the [cisco-network-node-utils](https://github.com/cisco/cisco-network-node-utils) git repository 
+
+Next install the code base. Clone the cisco-network-node-utils repo from your fork into a workspace:
+
+```bash
+git clone https://github.com/YOUR-USERNAME/cisco-network-node-utils.git
+cd cisco-network-node-utils/
+```
+
+As a best practice go ahead and create a topic/feature branch for your feature work using the `git branch feature/<feature_name>` command.
+
+```bash
+git branch feature/bash-shell
```

Comment:
oof. I changed all of the bash-shell references to 'tunnel' so you're going to have a bunch of merge conflicts.